code/datums/spells/devil.dm

```diff
@@ -0,0 +1,204 @@
+
+
+/obj/effect/proc_holder/spell/dumbfire/fireball/hellish
+	name = "Hellfire"
+	desc = "This spell launches hellfire at the target."
+
+	school = "evocation"
+	charge_max = 80
+	clothes_req = 0
+	invocation = "Your very soul will catch fire!"
+	invocation_type = "shout"
+	range = 2
+	action_background_icon_state = "bg_demon"
+
+/obj/effect/proc_holder/spell/turf/fireball/infernal/cast(var/turf/T)
+	explosion(T, -1, -1, -1, 4, 0, flame_range = 5)
+
+/obj/effect/proc_holder/spell/targeted/infernal_jaunt
+	name = "Infernal Jaunt"
+	desc = "Use hellfire to phase out of existence."
+	charge_max = 200
+	clothes_req = 0
+	selection_type = "range"
+	range = -1
+	cooldown_min = 0
+	overlay = null
+	include_user = 1
+	action_icon_state = "jaunt"
+	action_background_icon_state = "bg_demon"
+
+/obj/effect/proc_holder/spell/targeted/infernal_jaunt/cast(list/targets, mob/living/user = usr)
+	if(istype(user))
+		if(istype(user.loc, /obj/effect/dummy/slaughter/))
+			var/continuing = 0
+			if(istype(get_area(user), /area/shuttle/)) // Can always phase in in a shuttle.
+				continuing = 1
+			else
+				for(var/mob/living/C in orange(2, get_turf(user.loc))) //Can also phase in when nearby a potential buyer.
+					if (C.mind && C.mind.soulOwner == C.mind)
+						continuing = 1
+						break
+			if(continuing)
+				to_chat(user,"<span class='warning'>You are now phasing in.</span>")
+				if(do_mob(user,user,150))
+					user.infernalphasein()
+			else
+				to_chat(user,"<span class='warning'>You can only re-appear near a potential signer.")
+				revert_cast()
+				return ..()
+		else
+			user.notransform = 1
+			user.fakefire()
+			to_chat(user,"<span class='warning'>You begin to phase back into sinful flames.</span>")
+			if(do_mob(user,user,150))
+				user.infernalphaseout()
+			else
+				to_chat(user,"<span class='warning'>You must remain still while exiting.</span>")
+				user.ExtinguishMob()
+		start_recharge()
+		return
+	revert_cast()
+
+
+/mob/living/proc/infernalphaseout()
+	dust_animation()
+	spawn_dust()
+	src.visible_message("<span class='warning'>[src] disappears in a flashfire!</span>")
+	playsound(get_turf(src), 'sound/magic/enter_blood.ogg', 100, 1, -1)
+	var/obj/effect/dummy/slaughter/holder = PoolOrNew(/obj/effect/dummy/slaughter,loc)
+	src.ExtinguishMob()
+	if(buckled)
+		buckled.unbuckle_mob(src,force=1)
+	if(has_buckled_mobs())
+		unbuckle_all_mobs(force=1)
+	if(pulledby)
+		pulledby.stop_pulling()
+	if(pulling)
+		stop_pulling()
+	src.loc = holder
+	src.holder = holder
+	src.notransform = 0
+	fakefireextinguish()
+
+/mob/living/proc/infernalphasein()
+	if(src.notransform)
+		to_chat(src,"<span class='warning'>You're too busy to jaunt in.</span>")
+		return 0
+	fakefire()
+	src.loc = get_turf(src)
+	src.client.eye = src
+	src.visible_message("<span class='warning'><B>[src] appears in a firey blaze!</B>")
+	playsound(get_turf(src), 'sound/magic/exit_blood.ogg', 100, 1, -1)
+	addtimer(src, "fakefireextinguish", 15,TRUE)
```

Comment:
according to kor we had a very recent update that made it always be a better choice than spawn